### ISSUE TITLE
Bug > Carrier loss on Volla Phone

### ISSUE THREAD

**Issue by stekalb** [ubports/ubuntu-touch#1629]
- Device: Volla Phone
- Channel: developer
- Build: 20201215

### Steps to reproduce
I did not igure out how to reroduce

### Expected behavior
Device is connected to phone carrier if it displays a phone connection

### Actual behavior
Sometimes the connection to carrier is lost. But phone still displays connected.
It happens at least once per day.

### Logfiles and additional information
- Single SIM in Slot 1
- Country / Carrier: Switzerland / UPC
- Sometimes it's possible to restore the connection by restart ofono service
- most of time I need to reboot
- activating and deactivating flight mode does never restore the connection. Info displays usually "not connected" for network, but none the less "L" for data is displayed, but of course there is no data network without carrier
- In Volla TG group some other users also reported to be affected by this issue. It was questioned if that could be a GSI-specific-Issue

**Comment by stekalb** [comment]
[volla_syslog-snippet-20201219.log](https://github.com/ubports/ubuntu-touch/files/5723116/volla_syslog-snippet-20201219.log)
disconnect happend round about 22:40.
dmesg log is empty.

**Comment by fredldotme** [comment]
This issue seems to appear on all Halium 9 based ports, whether GSI or not. It's the same Binder "transaction failed" on at least the Pixel 3a and Pro1 as well.

**Comment by mgresko8** [comment]
Hello,

same here with Volla phone. I have never seen the issue on BQ Aquaris E4.5. When somebody calling me he hears the ring tone, but the phone is not ringing and no notification is seen. I cannot make calls. For second SIM, I am not reachable. After reboot everything works for some time.

Marek

**Comment by stylelibre** [comment]
For me, on my VollaPhone in Canada, I'm unable to connect to my carrier at all. If I flash with VollaOS, connection to carrier is working fine but on Ubuntu Touch, the connection to my carrier fail. I don't know if it was working previously since I only received my Volla last week. Same carrier is working fine on my Nexus5 (on UT) for many years (since UT is available on Nexus5). I tried reflashing with stable/rc/edge but they all fail :/.

I'd like to help with debugging if needed. I'll play around on my own but I'm new to this and I don't know where to start.

Cheers,
Fred

**Comment by Flohack74** [comment]
@stylelibre could it be that your carrier is VoLTE-only already? That would create issues for Ubuntu Touch....

**Comment by stylelibre** [comment]
@Flohack74 
Indeed, it was good tinking since VollaOS says VoLTE in the connection settings but my Nexus 5 with ut is working fine with the same carrier (ut says "LTE"). I tried to switch both sims (the one in VollaPhone and the one in Nexus 5) since the first one is much more recent but they act exactly the same (except for the different phone number).

Maybe the VollaPhone is only supporting VoLTE frequencies for my carrier (Koodo) and my Nexus 5 is using frequencies that is not only VoLTE. If VoLTE is not working in ut, that would explain the behavior I'm experiencing.

I'll go read on VoLTE, frequencies for my devices and carrier then try different things :). I'm starting to really like my VollaPhone. My main phone is an old Nexus 5 at the end of its life and it is harder and harder to find one in the second hand market. I also tried the PinePhone but it is not as good as the Volla. Not even close to a Nexus 5.

Many thanks for your answer.

**Comment by Flohack74** [comment]
Ah yes please compare the radio bands covered by each of them, it´s not about VoLTE but LTE bands in general and also GSM

**Comment by stylelibre** [comment]
But the same phone (VollaPhone) works well with a different OS (VollaOS based on Android9). I've check the frequencies used by my carrier and the supported one on the VollaPhone and I don't see any problem :/. I also tried SailfishOS on the VollaPhone but it doesn't boot after the installation (via UBports). I'll try SailfishOS later.

Also, on the VollaPhone with ut, I see "Unknown" under "SIM 1" ("SIM 2 has "No SIM" which is normal). Sometimes, just after a reboot and for a little while, I have "Denied" under "SIM 1". But my sim is not lock of anything :/

**Comment by JamiKettunen** [comment]
@stylelibre to install SFOS you need to have a system partition from Android (Volla OS flashed before starting), would be interesting to hear if it works there as the middleware (`ofono`) is reused on Ubuntu Touch as well.

**Comment by stylelibre** [comment]
OK, I've made a lot of tests. My first issue is definitly a frequency mismatch with my carrier. I've subscribe to another carrier with a new phone line and it works a bit more (I have voice and text). The data is not working (I'm roaming even if I not supposed to. My Nexus 5 on UT is not roaming with that SIM) but I think it's also a problem with frequencies. My conclusion is that the VollaPhone is really made for compatibility with the european market. I'll continue my tests but now I know it is not related to UT :)

Thanks for your help :)

**Comment by duportail** [comment]
Volla phone ubuports: several times per day that the phone does not receive calls or sms. No carrier on the two sim cards.
I use these sim cards in other phones (fairphone2, bq) without errors.
This volla phone is useless for me. Will have to send back all 6 pieces.

**Comment by duportail** [comment]
Is there a command to restart the service of the sim cards? Because after reboot all is ok for a while. No logs also.

**Comment by stekalb** [comment]
@duportail you can try the following command (either in terminal or ssh session) - that helps often but not everytime:
`sudo stop ofono`
`sudo start ofono`

**Comment by duportail** [comment]
here is the syslog, maybe it reveals some things. I am not a ubports expert.
https://www.ubuntushop.be/syslog

**Comment by duportail** [comment]
on the syslog, last call received was around 12:33h

**Comment by duportail** [comment]
just now again no simcards present. sudo stop ofono and sudo start ofono, gives me the simcards back.
Is there a possibilty to create script for crontab, check the sims and resart ofono if no sims active?

**Comment by ernesst** [comment]
> This issue seems to appear on all Halium 9 based ports, whether GSI or not. It's the same Binder "transaction failed" on at least the Pixel 3a and Pro1 as well.

Also on OP3 GSI, see some discussion. https://forums.ubports.com/topic/4481/halium9-gsi-oneplus-3-t/14?_=1612544026036
I've created a watchdog to restart ofono when a pattern is detected... not a great solution but I could use my phone daily. 
Somebody informed me that the problem tends to be similar on pinephone : https://gitlab.com/ubports/community-ports/pinephone/-/issues/229

**Comment by duportail** [comment]
The ofono check script is working. Not the best solution for normal users.
I am now not able to make a mobile internet connection on the volla phone with ubports stable or proposed release.

**Comment by Flohack74** [comment]
The root cause seems to be identified already, but we are not understanding why it happens and how to fix it. So please be still patient until we can provide a fix for that ;)

**Comment by AA303** [comment]
The same issue on EE in the UK. Usually fixed by switching between 2G/3G/4G and back, but not guaranteed. Sometimes rebooting the phone helps. I only become aware of the operator loss when I expect a call or SMS.

**Comment by AA303** [comment]
Update with observations: 

The loss of signal appears to happen overnight. Carrier loss may be happening, while the mobile phone remains connected to the WiFi. LTE indicator and network visually suggest that the signal is there. But when making a call, it fails. When rebooted Volla + UT, I get the message:

"No Signal" / "Searching".


This time to make the signal work, I had to switch-off for about 5 minutes and turn back on, which fixed the voice/SMS issue, but not data. Then I have tried "sudo reboot", which resolved the issue. I will remain off WiFi, to see if it makes any difference to carrier loss. 

What logs/where to find them to share here to help to identify the issue?

**Comment by AA303** [comment]
> root cause

What is it?

**Comment by duportail** [comment]
> The ofono check script is working. Not the best solution for normal users.
> I am now not able to make a mobile internet connection on the volla phone with ubports stable or proposed release.
Script not working for loosing carrier.
Mobile connection working if i enter apn settings mannualy:  telenetwap.be

**Comment by duportail** [comment]
last try to found solution i did this:
make system writeable with ubuntu touch tweak
in terminal:
sudo apt update
sudo apt-get dist-upgrade
because of an anbox related error:
sudo apt-get install -f

reboot.
did not have carrier loss since.

**Comment by francescoPadovani81** [comment]
Hello,
I've the same problem (with volla phone me too). This is the only issue which prevents me to use vollaphone+UT as my daily device ...and I would really like it to be that way.

@duportail , can I ask you which channel are you on? (I'm on stable/OTA-15). Is it safe to perform a dist-upgrade that way? Doesn't  it compromise future system updates? (sorry for my ignorance)

Just a note: some times (to date roughly twice a week) the problem for me shows in a bit worse way: the SIM indicator on the status bar completely disappears. Even the Cellular settings, under the Network section of system settings: fully vanished. The ofono restart doesn't do the trick in this case. I have to shutdown and then start again the phone (a simple reboot isnt' enough) to make the sim working again.

**Comment by duportail** [comment]
I am on stable ota-15.
Future updates? I do not know. But i have a working phone now,,,
Do not care about the inicators of the simms. I have one sim that is out of service according the indicator, but it is working, even mobile data. But if mobile connection is on, i do not see it on the top bar indicator.
I did the dist-upgrade, there was just a problem with some anbox package, so i did sudo apt-get install -f to correct this.
Also I had to reinstall dekko.
Do not forget to make your system writeable first. (ubuntu-tweak then system, make writeable once)

**Comment by greskom** [comment]
Hello,

I tested with mobile data off and wifi off. The issue triggered again. But I confirm it is far harder to trigger.

Even more I found out the wifi had major impact on battery life. When I have ifi on it last for max. 2 days. With wifi off I get get up to 5 days. The 5 years old bq aquaris E4.5 still lasts for 5 days. Sorry for off topic note.

Marek

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by AA303** [comment]
> Update with observations:
> 
> The loss of signal appears to happen overnight. Carrier loss may be happening, while the mobile phone remains connected to the WiFi. LTE indicator and network visually suggest that the signal is there. But when making a call, it fails. When rebooted Volla + UT, I get the message:
> 
> "No Signal" / "Searching".
> 
> This time to make the signal work, I had to switch-off for about 5 minutes and turn back on, which fixed the voice/SMS issue, but not data. Then I have tried "sudo reboot", which resolved the issue. I will remain off WiFi, to see if it makes any difference to carrier loss.
> 
> What logs/where to find them to share here to help to identify the issue?

Today I had experience signal-drop again, yet, visually I had all indicators suggested I had LTE available. I discovered this because I was expecting a call and when I tried calling back the attempt has failed. I have tested SMS, but the message got stuck. Pressing the power button and rebooting the phone has helped.

**Comment by AA303** [comment]
> last try to found solution i did this:
> make system writeable with ubuntu touch tweak
> in terminal:
> sudo apt update
> sudo apt-get dist-upgrade
> because of an anbox related error:
> sudo apt-get install -f
> 
> reboot.
> did not have carrier loss since.
> And the notifications on the screensaver working again!

I am trying these instructions and I will share my feedback in a few days.

**Comment by duportail** [comment]
Any news about a solution?

**Comment by AA303** [comment]
> Any news about a solution?

Your solution has worked for me. What exactly are you after?

**Comment by francescoPadovani81** [comment]
from me: 
I did nothing yet. I didn't follow the steps above: I just made a "sudo apt update", but not the "upgrade" (let alone the "dist-upgrade": yeah, I'm a coward). Nonetheless it's a week so far that it's working without problems. I think it's very hard to spot the trigger of such issue (and so debug). I'm definitely useless.
If there's any kind of log/command I can gather/execute in case it happens again:,let me know guys

**Comment by duportail** [comment]
Carrier connection still dropping. This phone with ubports is unusable. Cannot upgrade anymore also.

**Comment by AA303** [comment]
> > last try to found solution i did this:
> > make system writeable with ubuntu touch tweak
> > in terminal:
> > sudo apt update
> > sudo apt-get dist-upgrade
> > because of an anbox related error:
> > sudo apt-get install -f
> > reboot.
> > did not have carrier loss since.
> > And the notifications on the screensaver working again!
> 
> I am trying these instructions and I will share my feedback in a few days.

Yesterday as I was going on/off in the underground (reception was on/off) I have noticed the signal was lost and did not come back. I hope OTA-16 will address this issue.

**Comment by francescoPadovani81** [comment]
Yesterday, after a week that it has been working fine, it happens again. And just few minutes before it happened I noticed some switches between 2G/3G and LTE on my sim indicator on the upside bar. Not a full signal loss (I guess... but I'm not sure) as in case for @AA303, but I think nonetheless a possible trigger for this issue.
I had to restart the phone twice (after the first restart the SIM indicator and configurations were absent at all, as already described in a comment above).
When trying to perform a call (the way I noticed to be in the issue) on my syslog I saw the error
"ofono-ril-binder-plugin ofonod Call 1 ended with RIL cause 31 -> ofono reason 1"
Let's hope in OTA-16.

**Comment by NotKit** [comment]
For those who are testing on devel OTA channel, I pushed a workaround for one of the possible causes (binder transaction error in dmesg with code -28, as can be seen in posted syslog snipplet). It should be included since version 270.

If the issue can be still reproduced on latest devel, please post `/var/log/syslog` and output of `cat /sys/kernel/debug/binder/failed_transaction_log`.

**Comment by stekalb** [comment]
I am going to test it.

**Comment by ernesst** [comment]
> For those who are testing on devel OTA channel, I pushed a workaround for one of the possible causes (binder transaction error in dmesg with code -28, as can be seen in posted syslog snipplet). It should be included since version 270.
> 
> If the issue can be still reproduced on latest devel, please post `/var/log/syslog` and output of `cat /sys/kernel/debug/binder/failed_transaction_log`.

Could you link the commit ?
I would like to try it on my OP3 gsi, would you have a link to a deb package ?

**Comment by duportail** [comment]
> For those who are testing on devel OTA channel, I pushed a workaround for one of the possible causes (binder transaction error in dmesg with code -28, as can be seen in posted syslog snipplet). It should be included since version 270.
> 
> If the issue can be still reproduced on latest devel, please post `/var/log/syslog` and output of `cat /sys/kernel/debug/binder/failed_transaction_log`.

Installed it yesterday on the volla phone and did not have carrier loss since.

Guy

**Comment by JamiKettunen** [comment]
For people curious, the commit that seems to have helped here is [(halium) binder: do not use a separate queue for async oneway transactions](https://github.com/HelloVolla/android_kernel_volla_mt6763/commit/ade5d9861046a6f334c5753303eb329b5eb12289) on the Volla Phone kernel tree.

**Comment by duportail** [comment]
can this be added in the stable branch, so i can sell our phones in stock?

guy

**Comment by stekalb** [comment]
> I am going to test it.

So far stable. Runs since Wednesday without any issues (uptime 53h). I received and made several calls as well as SMS. 

I try not to reboot my device until Sunday evening and report once again after that. (Planned a hike on Sunday, so there will be various switches between 2G, 3G, 4G and no connection at all - the last test I did under these conditions I had to reboot twice to reestablish connectability. That was two weeks ago.)

**Comment by Flohack74** [comment]
@duportail unfortunately this fix came during the OTA freeze period, where we do not push any fixes anymore through to stable, to prevent stability issues. I will talk with the team if this is a critical fix we want to have for this OTA already, but I am not sure this will happen.

**Comment by stekalb** [comment]
> @duportail unfortunately this fix came during the OTA freeze period, where we do not push any fixes anymore through to stable, to prevent stability issues. I will talk with the team if this is a critical fix we want to have for this OTA already, but I am not sure this will happen.

That would be unfortunate in my opinion. At least the volla phones are hardly usable as daily driver without that fix if you need connectivity besides wifi.

**Comment by duportail** [comment]
> > @duportail unfortunately this fix came during the OTA freeze period, where we do not push any fixes anymore through to stable, to prevent stability issues. I will talk with the team if this is a critical fix we want to have for this OTA already, but I am not sure this will happen.
> 
> That would be unfortunate in my opinion. At least the volla phones are hardly usable as daily driver without that fix if you need connectivity besides wifi.

That is correct. 
This is a crucial function of a phone: receiving calls.

**Comment by JamiKettunen** [comment]
> @duportail unfortunately this fix came during the OTA freeze period, where we do not push any fixes anymore through to stable, to prevent stability issues. I will talk with the team if this is a critical fix we want to have for this OTA already, but I am not sure this will happen.

@Flohack74 I'd say this fixes a pretty critical current issue with the port. I've been running the patched kernel on my Volla for a week of uptime now and I've yet to have the cellular network stop functioning even once; calls, SMS, MMS (receiving; [sending issue appears to be unrelated](https://github.com/HelloVolla/ubuntu-touch-beta-tests/issues/94)) and cellular data are still all working perfectly fine without needing a reboot for a week.

**Comment by stekalb** [comment]
> > I am going to test it.
> 
> So far stable. Runs since Wednesday without any issues (uptime 53h). I received and made several calls as well as SMS.
> 
> I try not to reboot my device until Sunday evening and report once again after that. (Planned a hike on Sunday, so there will be various switches between 2G, 3G, 4G and no connection at all - the last test I did under these conditions I had to reboot twice to reestablish connectability. That was two weeks ago.)

Test finished. No connection issues at all whilst or after the hike. Many thanks for the fix/workaround! Much appreciated.

**Comment by Flohack74** [comment]
Seems I was not informed well enough, it seems to have been committed for the RC of OTA-16 and should therefore also land in stable. You could confirm by switching to RC channel and see if it works there still ;)

**Comment by TartanSpartan** [comment]
I'm so glad this fix will be landing in time for my Volla's delivery or close enough!

**Comment by AA303** [comment]
I am testing OTA-16. Today is day 1.

**Comment by greskom** [comment]
I am just trying workaround with sencond SIM manually set to 2G and wifi on.. It seems promising. I have not got the issue for several days.

Marek

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by AA303** [comment]
> I am testing OTA-16. Today is day 1.

Update: More than 30 days - OTA-16 has been working like a charm!

**Comment by verretor** [comment]
> @Flohack74
> Indeed, it was good tinking since VollaOS says VoLTE in the connection settings but my Nexus 5 with ut is working fine with the same carrier (ut says "LTE"). I tried to switch both sims (the one in VollaPhone and the one in Nexus 5) since the first one is much more recent but they act exactly the same (except for the different phone number).
> 
> Maybe the VollaPhone is only supporting VoLTE frequencies for my carrier (Koodo) and my Nexus 5 is using frequencies that is not only VoLTE. If VoLTE is not working in ut, that would explain the behavior I'm experiencing.
> 
> I'll go read on VoLTE, frequencies for my devices and carrier then try different things :). I'm starting to really like my VollaPhone. My main phone is an old Nexus 5 at the end of its life and it is harder and harder to find one in the second hand market. I also tried the PinePhone but it is not as good as the Volla. Not even close to a Nexus 5.
> 
> Many thanks for your answer.

I'm also a Koodo customer with a Vollaphone with Ubuntu Touch. I was able to use Koodo's internet but calls and text messages didn't work.

I haven't tested VollaOS but I suspect it would work since it worked for you.

**Comment by verretor** [comment]
> @Flohack74
> Indeed, it was good tinking since VollaOS says VoLTE in the connection settings but my Nexus 5 with ut is working fine with the same carrier (ut says "LTE"). I tried to switch both sims (the one in VollaPhone and the one in Nexus 5) since the first one is much more recent but they act exactly the same (except for the different phone number).
> 
> Maybe the VollaPhone is only supporting VoLTE frequencies for my carrier (Koodo) and my Nexus 5 is using frequencies that is not only VoLTE. If VoLTE is not working in ut, that would explain the behavior I'm experiencing.
> 
> I'll go read on VoLTE, frequencies for my devices and carrier then try different things :). I'm starting to really like my VollaPhone. My main phone is an old Nexus 5 at the end of its life and it is harder and harder to find one in the second hand market. I also tried the PinePhone but it is not as good as the Volla. Not even close to a Nexus 5.
> 
> Many thanks for your answer.

I'm also a Koodo customer with a Vollaphone with Ubuntu Touch. I managed to use Koodo's internet and text messages but calls didn't work.

**Comment by Flohack74** [comment]
Just saying that there is not such a thing as "VoLTE" frequencies. VoLTE stands for voice over LTE, means you can make calls via an LTE connection and do not need to fall back to 3G. So, the Vollaphone can or cannot work with LTE for certain carriers, depending on the frequencies used for LTE in your area. It seems to support those:
![grafik](https://user-images.githubusercontent.com/16186470/119476944-d430d180-bd4e-11eb-95e2-92c3b2f1c5ad.png)

**Comment by stylelibre** [comment]
I can confirm that the problem is with the frequencies. The Vollaphone is an European phone and the supported frequencies don't "cover" large enought for our frequencies (at least with the carriers I tried). Data and text are working because they are not using the same frequencies as the voice is using. At first I tought that it was VoLTE related but, as Flohack74 explained, VoLTE is software related, not hardware related. After a lot of try/error with the Vollaphone, I finally switch to a Pixel 3a which is working fine.

**Comment by verretor** [comment]
> I can confirm that the problem is with the frequencies. The Vollaphone is an European phone and the supported frequencies don't "cover" large enought for our frequencies (at least with the carriers I tried). Data and text are working because they are not using the same frequencies as the voice is using. At first I tought that it was VoLTE related but, as Flohack74 explained, VoLTE is software related, not hardware related. After a lot of try/error with the Vollaphone, I finally switch to a Pixel 3a which is working fine.

Didn't you say things worked with VollaOS though?

**Comment by stylelibre** [comment]
Yes because VollaOS support Voice over LTE and LTE (data) is working on Kodoo network with the Vollaphone. If I remember correctly, it is also working with Fizz. As far as I know, Ubuntu Touch don't have VoLTE implemented yet so the data and text are working but not the voice :/.

I'll do some testing from time to time with my Vollaphone to see if there is some progress. I really like the Vollaphone and I would love to make it my daily driver but for now, Pixel 3a it is :/.

**Comment by Flohack74** [comment]
Its weird, text should also go via 3G, same as voice, so why that works is a mystery then, but ok ^^ Yes the disagreement between NA and EU about technology and frequencies has a long history, I was also carrying 3 phones during a business trip in the Cincinnati, OH region an none of them could get a signal there. Was offline for days, then bought a Verizon prepaid only to find out that they whitelist devices and none of my devices was in .. graaah

**Comment by Flohack74** [comment]
Finally found one tiny spot besides the road where I could make and receive calls... as uncomfortable it was, but it was ok then ;)

**Comment by verretor** [comment]
> Its weird, text should also go via 3G, same as voice, so why that works is a mystery then, but ok ^^ Yes the disagreement between NA and EU about technology and frequencies has a long history, I was also carrying 3 phones during a business trip in the Cincinnati, OH region an none of them could get a signal there. Was offline for days, then bought a Verizon prepaid only to find out that they whitelist devices and none of my devices was in .. graaah

I'm confused about it too. I could send text messages to my friend but couldn't call anyone.

I will have to keep digging but since it works with VollaOS, I suspect it has to do with Ofono. Maybe there's a way to use Sailfish OS's Ofono in a Chroot or some other trick.

**Comment by Flohack74** [comment]
Dear all, can you please test this with latest RC __before__ we call for public testing, I want to make sure that this is not an immediate fail. Thanks!

**Comment by stylelibre** [comment]
Hello, 
I tried with the current "stable" and was able to text (send and receive) and make phone call but was not able to receive phone call. So I changed the channel to "release candidate" and after the installation of the rc version, nothing changed. Everything was the same (carrier wise). Text both ways, able to make calls and unable to receive calls. I did not notice any regression in the few hours I've tested it.

I tried all that with different carriers. I'm pretty sure that since the phone (Volla) is for european market, I'm also experimenting limitations with the frequencies available to me (north of Montreal). 

The VollaPhone I have is now used by a kid who doesn't need cellular connectivity but I'm pretty sure that, at some point, this kid will want to be able to text and call so it would be nice not to have to find another phone :).

Thanks for the progress on this.
Cheers,
Fred

**Comment by stekalb** [comment]
No issues so far on my side.

**Comment by Flohack74** [comment]
@stekalb you are on a Vollaphone?

**Comment by stekalb** [comment]
Still on volla, yess

> @stekalb you are on a Vollaphone?

**Comment by Flohack74** [comment]
Ok I will close this, if there is any further problems here, please open a new ticket.
